src/playground/services/invoices/delete/endpoint.clj

```diff
@@ -1,17 +1,27 @@
 (ns playground.services.invoices.delete.endpoint
   (:require
+   [clojure.spec.alpha :as spec]
    [clojure.java.jdbc :as jdbc]
    [honeysql.core :as h]
-   [playground.services.invoices.delete.logic :as logic]))
+   [playground.services.invoices.delete.logic :as logic]
+   [playground.services.invoices.retrieve-all.logic :as retrieve-all.logic]
+   [playground.services.invoices.delete.view :as view]))
 
-(defn perform [{:keys [db] :as request}]
+(spec/def ::id nat-int?)
+
+(spec/def ::api (spec/keys :req-un [::id]))
+
+(defn perform [{{:keys [id]} :path-params :keys [db] :as request}]
   (let [db (->> db :pool (hash-map :datasource))
-        _  (->> (logic/to-delete)
+        _  (->> (logic/to-delete id)
                 (h/format)
                 (jdbc/execute! db))
-        response (->> (logic/to-serialize)
-                      (h/format)
-                      (jdbc/query db)
-                      (first))]
+        fetch  (h/format (retrieve-all.logic/query-all))
+        left   (->> (logic/to-serialize)
+                    (h/format)
+                    (jdbc/query db)
+                    (first))
+        result (jdbc/query db fetch)
+        result-map {:result result :deleted id}]
     {:status 200
```

Comment:
Instead of rendering a view from the `delete` endpoint, respond with `:status 302`, redirecting to the `retrieve-all` page. It's how "create", "delete" and "update" actions normally work. Only "show", "edit", and "list" actions should render any items at all.

(if the edit vs update distinction seems confusing, see the Rails guides)

Accordingly, you have to make the `retrieve-all` able to display a conditional message. That is called a flash in Rails.

Usage is described [here](https://api.rubyonrails.org/classes/ActionDispatch/Flash.html) and this is how the user perceives it:

![image](https://user-images.githubusercontent.com/1162994/54859733-35912c00-4d11-11e9-847e-1266e3c6f67f.png)

(the red thingy)

---

For this PR, just do the 302 _without_ implementing a flash.

As your next PR, learn how to add/implement a flash in Pedestal. It may be slightly hard. Make sure to work cleanly, in small steps. Use the extended commit message to document your thinking and findings:

![image](https://user-images.githubusercontent.com/1162994/54859767-8d2f9780-4d11-11e9-8529-80d6d9591092.png)

^ I'd fill this commit UI like this:

![image](https://user-images.githubusercontent.com/1162994/54859790-d1bb3300-4d11-11e9-9888-a5a9f849b460.png)

For harder problems is _really_ important that you work in that manner. That way both you and me will be able to understand your thinking today and tomorrow.

Finally, please improve the form of your commit messages.

Please take inspiration from here: https://github.com/zenmacs/.emacs.d/commits/master

* Start with a verb
* Use present simple
* Use backticks
* When needed, add a extended message